src/library/scala/collection/Iterator.scala

```diff
@@ -144,7 +144,7 @@ trait Iterator[+A] extends IterableOnce[A] with IterableOnceOps[A, Iterator, Ite
    *
    *  Typical uses can be achieved via methods `grouped` and `sliding`.
    */
-  class GroupedIterator[B >: A](self: Iterator[B], size: Int, step: Int) extends AbstractIterator[immutable.Seq[B]] {
+  class GroupedIterator[B >: A]private[Iterator] (self: Iterator[B], size: Int, step: Int) extends AbstractIterator[immutable.Seq[B]] {
```

Comment:
I can't think of a compelling use case for calling the constructor directly but we're exposing the type anyway. Hiding the constructor wouldn't really help.